VegProcessor/species_hsi/catfish.py

```diff
@@ -0,0 +1,785 @@
+            # Create console handler and set level
+            ch = logging.StreamHandler()
+            ch.setLevel(logging.INFO)
+
+            # Create formatter and add it to the handler
+            formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+            )
+            ch.setFormatter(formatter)
+
+            # Add the handler to the logger
+            self._logger.addHandler(ch)
+
+    def calculate_si_1(self) -> np.ndarray:
+        """Percent pools during average summer flow"""
+        self._logger.info("Running SI 1")
+        si_1 = self.template.copy()
+
+        if self.v1_pct_pools_avg_summer_flow is None:
+            self._logger.info(
+                "Percent pools during avg summer flow is not provided. Setting index to 1."
+            )
+            si_1[~np.isnan(si_1)] = 1
+
+        else: 
+            # condition 1
+            mask_1 = (self.v1_pct_pools_avg_summer_flow < 40)
+            si_1[mask_1] = (
+                0.02 * (self.v1_pct_pools_avg_summer_flow [mask_1])
+            ) + 0.2 
+            
+            # condition 2
+            mask_2 = (
+                (self.v1_pct_pools_avg_summer_flow >= 40) & 
+                (self.v1_pct_pools_avg_summer_flow <= 60)
+            )
+            si_1[mask_2] = 1
+
+            # condition 3
+            mask_3 = (
+                (self.v1_pct_pools_avg_summer_flow > 60) & 
+                (self.v1_pct_pools_avg_summer_flow <= 100)
+            )
+            si_1[mask_3] = (
+                -0.0125 * (self.v1_pct_pools_avg_summer_flow [mask_3])
+            ) + 1.75 
+
+        if np.any(np.isclose(si_1, 999.0, atol=1e-5)):
+            raise ValueError("Unhandled condition in SI logic!")
+
+        return self.clip_array(si_1)
+
+    def calculate_si_2(self) -> np.ndarray:
+        """Percent cover (logs, boulders, cavities, brush, debris, or standing timber) 
+        during summer within pools, backwater areas """
+        self._logger.info("Running SI 2")
+        si_2 = self.template.copy()
+
+        if self.v2_pct_cover_in_summer_pools_bw is None:
+            self._logger.info(
+                "Pct cover during summer within pools, backwaters is not provided. ""Setting index to 1."
+            )
+            si_2[~np.isnan(si_2)] = 1
+
+        else:
+            # condition 1
+            mask_1 = (
+                (self.v2_pct_cover_in_summer_pools_bw >= 0) & 
+                (self.v2_pct_cover_in_summer_pools_bw < 40)
+            )
+            si_2[mask_1] = (
+                0.022 * (self.v2_pct_cover_in_summer_pools_bw [mask_1])
+            ) + 0.1397
+            
+            # condition 2
+            mask_2 = (
+                (self.v2_pct_cover_in_summer_pools_bw >= 40)
+            )
+            si_2[mask_2] = 1
+    
+        if np.any(np.isclose(si_2, 999.0, atol=1e-5)):
+            raise ValueError("Unhandled condition in SI logic!")
+        
+        return self.clip_array(si_2)
```

Comment:
Looks like the slides skip si_3,  Maybe a good idea to add a placeholder so we know this is correct and not an error.
```
def calculate_si_3(self):
    """No logic exists for si_3.
    """
    return NotImplementedError
```